upgrade/update-repository.py

```diff
@@ -0,0 +1,345 @@
+def load_transformers(graphdb_info):
+    # Get the list of available transformations.
+
+    plugins_subdirs = os.listdir("plugins")
+    pr_nums = []
+
+    for dirname in plugins_subdirs:
+        match = plugin_dirname_regex.match(dirname)
+
+        if match is not None:
+            pr_nums.append(int(match.group(1)))
+
+    # Get the version string attached to knora-base in the repository.
+
+    knora_base_pr_num_sparql = """
+        PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
+
+        SELECT ?ontologyVersion
+        WHERE {
+            <http://www.knora.org/ontology/knora-base> knora-base:ontologyVersion ?ontologyVersion .
+        }
+    """
+
+    query_result_rows = rdftools.do_select_request(graphdb_info=graphdb_info, sparql=knora_base_pr_num_sparql)
+
+    # Did we find a version string?
+    if len(query_result_rows) > 0:
+        # Yes. Parse it.
+
+        ontology_version_string = query_result_rows[0]["ontologyVersion"]["value"]
+        match = knora_base_version_string_regex.match(ontology_version_string)
+
+        if match is None:
+            raise rdftools.UpdateException("Could not parse knora-base:ontologyVersion: {}".format(ontology_version_string))
+
+        print("Repository version: {}".format(ontology_version_string))
+        repository_pr_num = int(match.group(1))
+    else:
+        # No. Run all available transformations.
+        repository_pr_num = 0
+        print("Repository has no knora-base:ontologyVersion.")
+
+    # Make a sorted list of transformations needed for this repository.
+    needed_pr_nums = sorted(list(filter(lambda pr_num: pr_num > repository_pr_num, pr_nums)))
+
+    # If the list of transformations is empty, do nothing.
+    if len(needed_pr_nums) == 0:
+        return []
+
+    needed_pr_nums_str = ', '.join(map(str, needed_pr_nums))
+    print("Required updates: " + needed_pr_nums_str)
+    transformers = []
+
+    # Load a GraphTransformer instance for each transformation.
+    for transformer_pr_num in needed_pr_nums:
+        # Load the transformer's module.
+        transformer_module = importlib.import_module("plugins.pr{}.update".format(transformer_pr_num))
+
+        # Get the transformer class definition from the module.
+        transformer_class = getattr(transformer_module, "GraphTransformer")
+
+        # Make an instance of the transformer class.
+        transformer = transformer_class()
+
+        # Add its PR number to it.
+        transformer.pr_num = transformer_pr_num
+
+        # Add it to the list.
+        transformers.append(transformer)
+
+    return transformers
+
+
+# Command-line invocation.
+def main():
+    default_graphdb_host = "localhost"
+    default_repository = "knora-test"
+
+    parser = argparse.ArgumentParser(description="Updates a Knora repository.")
+    parser.add_argument("-g", "--graphdb", help="GraphDB host (default '{}')".format(default_graphdb_host), type=str)
```

Comment:
You can add default values with the `default` parameter. Here is an example:

https://github.com/dasch-swiss/0807-mls-scripts/blob/7e3623b623a347ea254f90807dec17c5a34a7d17/mls-import.py#L559

Reply:
Thanks, done in e39736c.